src/library/scala/collection/immutable/Range.scala

```diff
@@ -373,11 +373,11 @@ sealed abstract class Range(
     if (x == end && !isInclusive) false
     else if (step > 0) {
       if (x < start || x > end) false
-      else (step == 1) || (((x - start) % step) == 0)
+      else (step == 1) || (Integer.remainderUnsigned(x - start, step) == 0)
     }
     else {
       if (x < end || x > start) false
-      else (step == -1) || (((x - start) % step) == 0)
+      else (step == -1) || (Integer.remainderUnsigned(x - start, step) == 0)
```

Comment:
In this case, `x - start` and `step` are `< 0`. That was fine with the signed `%`, but with unsigned remainder, we need to make sure to pass in non-negative values:
```suggestion
      else (step == -1) || (Integer.remainderUnsigned(start - x, -step) == 0)
```

Reply:
Thanks a lot, and I missed this part in my test.